src/sirhelpers.c

```diff
@@ -39,7 +39,7 @@ bool __sir_validptr(const void* restrict p, bool fail) {
     bool valid = NULL != p;
     if (!valid && fail) {
         (void)_sir_seterror(_SIR_E_NULLPTR);
-        SIR_ASSERT(!"NULL pointer");
+        SIR_ASSERT(!(bool)"NULL pointer!");
```

Comment:
I see you did this in multiple places. What's the warning?

Reply:
```sh
src/sirhelpers.c:42:21: error: implicit conversion turns string literal into bool: 'char[14]' to 'bool' [-Werror,-Wstring-conversion]
        SIR_ASSERT(!"NULL pointer!");
                   ~^~~~~~~~~~~~~~~
```